src/Tools/AI Test Toolkit/src/TestSuite/AITALTestSuiteMgt.Codeunit.al

```diff
@@ -193,4 +193,59 @@ codeunit 149037 "AIT AL Test Suite Mgt"
                 Error(NoTestOutputFoundErr);
         end;
     end;
+
+    /// <summary>
+    /// Import the Test Input Dataset from an InStream of a dataset in a supported format.
+    /// Overwrite the dataset if the dataset with same filename is already imported by the same app
+    /// Error if the dataset with the same filename is created by a different app
+    /// </summary>
+    /// <param name="DatasetFileName">The file name of the dataset file which will be used in the description of the dataset.</param>
+    /// <param name="DatasetInStream">The InStream of the dataset file.</param>
+    procedure ImportTestInputs(DatasetFileName: Text; var DatasetInStream: InStream)
+    var
+        TestInputGroup: Record "Test Input Group";
+        TestInputsManagement: Codeunit "Test Inputs Management";
+        CallerModuleInfo: ModuleInfo;
+        EmptyGuid: Guid;
+        SameDatasetNameErr: Label 'The test input dataset %1 with the same file name already exists. The dataset was uploaded %2. Please rename the current dataset or delete the existing dataset.', Comment = '%1 = test input dataset Name, %2 = "from the UI" or "by the app id: {app_id}';
+        SourceOfTheDatasetIsUILbl: Label 'from the UI';
+        SourceOfTheDatasetIsAppIdLbl: Label 'by the app id: %1', Comment = '%1 = app id';
+    begin
+        // Check if the dataset with the same filename exists
+        NavApp.GetCallerModuleInfo(CallerModuleInfo);
+        TestInputGroup.SetLoadFields("File Name", "Imported by AppId");
+        TestInputGroup.SetRange("File Name", DatasetFileName);
+
+        if TestInputGroup.FindFirst() then
+            if TestInputGroup."Imported by AppId" = CallerModuleInfo.Id then
+                TestInputGroup.Delete(true) // Overwrite the dataset
+            else
+                case TestInputGroup."Imported by AppId" of
```

Comment:
Minor - I would move this to a method and use if statements. Case is odd if there is no enum

Reply:
Running into AA0022 Error: https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0022?wt.mc_id=d365bc_inproduct_alextension

With my current settings, won't fix.